parachain/pallets/ethereum-light-client/src/lib.rs

```diff
@@ -265,6 +265,48 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		/// Reset's the internal state of the light client.
+		///
+		/// `forked_at` - The hash of the header where the fork originated. The common ancestor.
+		///
+		/// This is only meant to be used in staging environments where long range forks
+		/// can happen.
+		///
+		/// Requires sudo user.
+		#[pallet::weight(1_000_000)]
+		#[transactional]
+		pub fn handle_long_range_fork(origin: OriginFor<T>, forked_at: H256) -> DispatchResult {
+			ensure_root(origin)?;
+
+			let stored_header = <Headers<T>>::get(forked_at).ok_or(Error::<T>::MissingHeader)?;
+			let header = stored_header.header;
+
+			let required_descendants = T::DescendantsUntilFinalized::get() as usize;
+			let best_block_id = EthereumHeaderId { number: header.number, hash: forked_at };
+
+			// Return error if the fork occurs after the current finalized block
+			let current_finalized = <FinalizedBlock<T>>::get();
+			if best_block_id.number >= current_finalized.number {
+				return Err(Error::<T>::Unknown.into())
+			}
+
+			// Return error if the fork does not have the required number of decendants
+			let (new_finalized_hash, _) =
+				Self::find_finalized_decendant(&best_block_id, required_descendants)
+					.ok_or(Error::<T>::Unknown)?;
+
+			let finalized_stored_header =
+				<Headers<T>>::get(new_finalized_hash).ok_or(Error::<T>::MissingHeader)?;
+			let header = finalized_stored_header.header;
+			let finalized_block_id =
+				EthereumHeaderId { number: header.number, hash: new_finalized_hash };
+
+			<FinalizedBlock<T>>::put(finalized_block_id);
+			<BestBlock<T>>::put((best_block_id, header.difficulty));
```

Comment:
Looks like a bug as `header.difficulty` is the difficulty of the new finalized block, not the new best block.

Reply:
Is it the sum of difficulty between the best and finalized block?

Reply:
Actually looks like its the total difficulty (sum of all individual block difficulties since genesis) of the chain. So you're gonna need to use some of the logic from import_validated_header: https://github.com/Snowfork/snowbridge/blob/main/parachain/pallets/ethereum-light-client/src/lib.rs#L361

Reply:
Awesome 👍